### PR TITLE
Change processor name from x86 to i686 for 32bit host

### PR DESCRIPTION
**Pull request by seanshpark**
CMake document says name is `uname -p` and for my 32bit Ubuntu Linux  says `i686`
https://cmake.org/cmake/help/v2.8.12/cmake.html#variable:CMAKE_SYSTEM_PROCESSOR

This will fix https://github.com/Samsung/iotjs/issues/287

### CODE REVIEW

File: jerry-libc/CMakeLists.txt [hunk 1]
```diff
@@ -34,8 +34,8 @@ set(COMPILE_FLAGS_LIBC "${COMPILE_FLAGS_JERRY} ${C_FLAGS_JERRY}")
    set(DEFINES_LIBC_ARMV7_HF __TARGET_HOST_ARMv7 __TARGET_HOST_ARMv7_HARD_FLOAT)
   # ARMv7-el
    set(DEFINES_LIBC_ARMV7_EL __TARGET_HOST_ARMv7 __TARGET_HOST_ARMv7_SOFT_FLOAT)
-  # x86
-   set(DEFINES_LIBC_X86 __TARGET_HOST_x86)
+  # i686
+   set(DEFINES_LIBC_I686 __TARGET_HOST_i686)
```
Comment (egavrin) [inline jerry-libc/CMakeLists.txt]: IMO we shouldn't rename this define.

Comment (seanshpark) [inline jerry-libc/CMakeLists.txt]: OK, I think I know what you mean :)

**Comment by seanshpark** [comment]
Also removed some trailing white spaces.

**Comment by seanshpark** [comment]
Anyone interested in reviewing this? :)

**Comment by egavrin** [comment]
Yes, sorry ^_^

**Comment by egavrin** [comment]
Please fix `toolchain_afl_i686.cmake` config too.

**Comment by egavrin** [comment]
LGTM after fix.

**Comment by seanshpark** [comment]
Fixed as you wish :)

**Comment by egavrin** [comment]
LGTM ^_^

**Comment by galpeter** [comment]
lgtm

**Pull request merged**